### PR TITLE
Fix calls to master from others ACs.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/4732/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -2694,6 +2700,26 @@ def remove_tq_endpoints
...
+  def write_tools_config
+    ["#{@options['keyname']}.secret",
+        "locations-#{@options['keyname']}.json"].each { |config|
+      # Read the current config file for the deployment
+      current = File.read("#{APPSCALE_CONFIG_DIR}/#{config}")
+
+      # Compare it with what the tools have and override if needed.
+      config_file = "#{APPSCALE_TOOLS_CONFIG_DIR}/#{config}"
+      tools_current = ''
+      tools_current = File.read(config_file) if File.exists?(config_file)
```
Comment (cdonati) [inline AppController/djinn.rb]: Usually a try+rescue is better for this case than a check+set.

File: AppController/djinn.rb [hunk 2]
```diff
@@ -2694,6 +2700,26 @@ def remove_tq_endpoints
     HAProxy.remove_tq_endpoints
   end
 
+  # TODO: this is a temporary. The dependency to the tools should be
+  # removed.
+  def write_tools_config
+    ["#{@options['keyname']}.secret",
+        "locations-#{@options['keyname']}.json"].each { |config|
+      # Read the current config file for the deployment
+      current = File.read("#{APPSCALE_CONFIG_DIR}/#{config}")
```
Comment (cdonati) [inline AppController/djinn.rb]: This should probably rescue ENOENT so that it can't crash the controller.

File: AppController/djinn.rb [hunk 3]
```diff
@@ -2694,6 +2700,26 @@ def remove_tq_endpoints
     HAProxy.remove_tq_endpoints
   end
 
+  # TODO: this is a temporary. The dependency to the tools should be
+  # removed.
+  def write_tools_config
+    ["#{@options['keyname']}.secret",
+        "locations-#{@options['keyname']}.json"].each { |config|
```
Comment (cdonati) [inline AppController/djinn.rb]: I believe aligning the elements is the convention for indenting here.

File: AppController/djinn.rb [hunk 4]
```diff
@@ -2694,6 +2700,26 @@ def remove_tq_endpoints
...
+      # Read the current config file for the deployment
+      current = File.read("#{APPSCALE_CONFIG_DIR}/#{config}")
+
+      # Compare it with what the tools have and override if needed.
+      config_file = "#{APPSCALE_TOOLS_CONFIG_DIR}/#{config}"
+      tools_current = ''
+      tools_current = File.read(config_file) if File.exists?(config_file)
+      if tools_current != current
+        FileUtils.mkdir_p(APPSCALE_TOOLS_CONFIG_DIR)
+        File.open(config_file, 'w+') { |dest_file| dest_file.write(current) }
```
Comment (cdonati) [inline AppController/djinn.rb]: 'w' should be sufficient here.

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/4737/

**Pull request merged**